theano/scan_module/scan_opt.py

```diff
@@ -1337,11 +1337,10 @@ def sanitize(x):
                 for idx, val in enumerate(store_steps[op.n_mit_mot:]):
                     if val == 0:
                         if idx < op.n_mit_sot + op.n_sit_sot:
-                            _nw_input = nw_inputs[offset + idx].owner.inputs[1]
-                            nw_input = scan_utils.expand(_nw_input, nw_steps)
-                            nw_inputs[offset + idx] = nw_input
```

Comment:
If I understand correctly, this essentially disables the part of the optimization that replaces the input buffer with a smaller one, in all cases.
I think it would be better to check if the buffer has other clients apart from the scan, and only skip that part in that case.